### DISCUSSION TITLE
Rapid v2.0 beta is here! ❤️

### DISCUSSION THREAD

**Discussion by Bonkles** [facebook/Rapid#738] [Show and tell]
##  Introducing - the Rapid v2.0 public beta

> &nbsp;
> 👉 &nbsp;   https://mapwith.ai/rapid-v2-beta
> &nbsp;

### What is this all about? 

A much faster Rapid!  We originally announced this work [here](https://github.com/facebookincubator/RapiD/discussions/481). Previous versions of Rapid used SVG (scalable vector graphics) to draw all of the map features in your web browser.  It worked, but if you pushed more map data at it, the renderer would slow to a crawl.  So we rewrote the renderer to instead use WebGL using the Pixi.js rendering library.  WebGL is a much faster way to render complex graphics, using your computer’s dedicated GPU to draw to the web browser.  This means **Rapid v2** performs much better and stays snappy and usable in densely mapped places like Manhattan, São Paulo, or Tokyo. 

### How can I use it?

Visit https://mapwith.ai/rapid-v2-beta, use it to edit OpenStreetMap, and submit any bugs or feedback you have. We’ve improved the bug reporting functionality to auto-fill a bunch of helpful information. 

### How much faster is it?

_Lots._ Expect **Rapid v2** to deliver around a 5-10x framerate improvement over Rapid v1, with even greater gains in denser areas. Certain kinds of editing that were impossible before are now possible!  Panning, zooming, and drawing on the map feel much more responsive. 

_Location: Rapid City, South Dakota_

Rapid v1: 3 frames/sec 😭|Rapid v2: 30 frames/sec
---|---
![image14](https://user-images.githubusercontent.com/1887955/218803790-c18cc95a-653a-4af6-b60f-8eaef49b9a7f.gif)|![image12](https://user-images.githubusercontent.com/1887955/218803625-eeff7a59-92e0-490e-8203-4487aa000766.gif)

##### Is there anything else improved about it besides speed?
Yes! Here’s a short list of improvements: 

### More Data

There is always more data to map, and our world is being mapped in greater detail than ever before.  People are adding sidewalks, addresses, buildings, and trees to OpenStreetMap, and Rapid v1 struggled to keep up.  Sometimes it would just drop features and display a message like “700 hidden features”.  Rapid v2 doesn’t hide anything from you!

Rapid v1: “795 hidden features” 😭|Rapid v2: All buildings are shown
---|---
![image10](https://user-images.githubusercontent.com/1887955/218804374-f49313e8-5f58-4314-8912-15883edf7e7d.png)|![image3](https://user-images.githubusercontent.com/1887955/218804428-49034d02-96a3-4eb7-91a9-ddd65451703f.png)

Rapid v1: “1475 hidden features” 😭|Rapid v2: All trees are shown
---|---
![image13](https://user-images.githubusercontent.com/1887955/218804856-a9aa499e-6dcd-47d8-a0ce-e1c97e67be71.png)|![image2](https://user-images.githubusercontent.com/1887955/218804832-09042900-1dda-418f-8fa9-9a83d2394b19.png)

### Improved Labeling

We’ve reworked the code to place labels more efficiently, allowing for more labels on Points of Interest (POIs) like shops, restaurants, and businesses. 

We also now use a consistent approach to labeling POIs, regardless of whether they are mapped as points or as polygons.  If mapped as a polygon you’ll now see a “virtual” point placed within it.

Rapid v1: Inconsistent POI labeling|Rapid v2: All businesses labeled!
---|---
![image4](https://user-images.githubusercontent.com/1887955/218805324-7d489616-60ba-44b2-aa7a-07b04c2f2d7b.png)|![image9](https://user-images.githubusercontent.com/1887955/218805340-14eb7811-ec24-45bb-8f8c-01980e6a4b6d.png)

Additionally, we now render labels around the perimeter of area features, such as neighborhoods, waterways, golf courses, parks, and shopping plazas - so when you zoom in, you’ll have much more context about the places that you’re editing.

Rapid v1: Area labels missing or misplaced|Rapid v2: Areas labeled around edges
---|---
<img width="948" alt="image11" src="https://user-images.githubusercontent.com/1887955/218805700-f462cd77-69d5-4fb0-bfb0-a304c05fec49.png">|<img width="955" alt="image5" src="https://user-images.githubusercontent.com/1887955/218805726-7d96c66c-f8bc-4c9b-bf06-65174d50be84.png">

### Edit at lower zoom levels

Now that we can show more detail than ever before, we don’t cut off editing at zoom 16 like we used to. This also means that editing at higher latitudes will display more of the map - no more feeling artificially zoomed in while editing buildings in Norway, Siberia, or Chile. 😀 

Rapid v1: “Zoom in to edit” 😭|Rapid v2: Edit Tromsø at zoom 15
---|---
![image8](https://user-images.githubusercontent.com/1887955/218805997-23bb92cf-35ab-4892-b396-5dfb265a908c.png)|![image7](https://user-images.githubusercontent.com/1887955/218806012-1b69e975-f3a6-4b8c-ae9a-87d847fc4ff0.png)

### What do I do if I find a bug?

Simply click the bug nub in the bottom-right of the RapiD UI. All you need to do is type a bug description and how to reproduce.  _(Note: You will need a Github account to submit bug reports to RapiD)_

<img width="69" alt="image1" src="https://user-images.githubusercontent.com/1887955/218806121-2271070f-5631-42a5-b935-7bc3ad46d70e.png"><img width="101" alt="image6" src="https://user-images.githubusercontent.com/1887955/218806122-22609035-1e6a-4369-9fc2-ea576c7006c6.png">

We're available to chat!  Ping us on the `#mapwithai_feedback` channel on either:
* [OpenStreetMap US Slack](https://slack.openstreetmap.us/)
* [HOTOSM Slack](https://slack.hotosm.org/)

Happy Mapping!

**Comment by kechan000** [comment]
I think all this stuff is great, but the scroll wheel zooms up instead of in. The only way for this to work is using the scroll wheel while holding the shift button. I can't change this in settings, and I relied on easy zooming in and out and not holding the shift key while using it. Can RapiD v2 remove the scroll + shift to zoom in thing and instead just do the scroll wheel and nothing else?

**Reply by Bonkles** [reply]
We hear you! This was one of the biggest issues we heard about with the beta thus far. It's already fixed in in a refresher update we'll be pushing live in one or two days. 

You can read more about the discussion around the bug here: https://github.com/facebook/Rapid/issues/680

TL;DR: THere will be a setting/preference where you can alter this behavior.

**Comment by misibacsi** [comment]
I cannot close "What's new v2" window, now I cannot use RapiD (v1)
Yesterday it worked somehow.
Firefox latest.

**Reply by Bonkles** [reply]
Can you click the bug submit button in the lower right? that will give us an idea of the exact firefox version you're running- also, do you have any addons for firefox running?

**Reply by Bonkles** [reply]
Ah, we think we see the problem. Scroll is disabled so you can't scroll down to see the button/'don't show this again' checkbox. We'll roll out a fix soon.

**Reply by misibacsi** [reply]
It's OK now, thanks.
